src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -13,10 +13,13 @@ namespace Microsoft.PowerFx.Functions
     // Direct ports from JScript. 
     internal static partial class Library
     {
-        private static readonly object _randomizerLock = new object();
+        private static readonly IRandomService _defaultRandService = new DefaultRandomService();
 
-        [ThreadSafeProtectedByLock(nameof(_randomizerLock))]
-        private static Random _random;
+        private static T GetService<T>(this IServiceProvider services, T defaultService)
+        {
+            var service = (T)services.GetService(typeof(T));
+            return service ?? defaultService;
+        }
```

Comment:
This seems like a generally useful helper as we expose more services, is there a better place for it than LibraryMath.cs?